build/build_edge.sh

```diff
@@ -0,0 +1,182 @@
+}
+
+usage()
+{
+    echo "Usage: $PROGRAM [options] "
+    echo "  options:"
+    echo "          -help                           Print this usage"
+    echo "          -aarch                          Architecture <aarch32/aarch64>"
+    echo "          -clean, clean                   Remove build directories"
+    echo ""
+}
+
+usage_and_exit()
+{
+    usage
+    exit $1
+}
+
+# --- Internal funtions ---
+install_recipes()
+{
+    META_USER_PATH=$1
+
+    SAVED_OPTIONS_LOCAL=$(set +o)
+    set +e
+    mkdir -p ${META_USER_PATH}/recipes-xrt/xrt
+    mkdir -p ${META_USER_PATH}/recipes-xrt/zocl
+    XRT_BB=${META_USER_PATH}/recipes-xrt/xrt/xrt_git.bbappend
+    ZOCL_BB=${META_USER_PATH}/recipes-xrt/zocl/zocl_git.bbappend
+    grep "inherit externalsrc" $XRT_BB
+    if [ $? != 0 ]; then
+        echo "inherit externalsrc" > $XRT_BB
+        echo "EXTERNALSRC = \"$XRT_REPO_DIR/src\"" >> $XRT_BB
+        echo 'EXTERNALSRC_BUILD = "${WORKDIR}/build"' >> $XRT_BB
+        echo 'PACKAGE_CLASSES = "package_rpm"' >> $XRT_BB
+    fi
+
+    grep "inherit externalsrc" $ZOCL_BB
+    if [ $? != 0 ]; then
+        echo "inherit externalsrc" > $ZOCL_BB
+        echo "EXTERNALSRC = \"$XRT_REPO_DIR/src/runtime_src/core/edge/drm/zocl\"" >> $ZOCL_BB
+        echo "EXTERNALSRC_BUILD = \"$XRT_REPO_DIR/src/runtime_src/core/edge/drm/zocl\"" >> $ZOCL_BB
+        echo 'PACKAGE_CLASSES = "package_rpm"' >> $ZOCL_BB
+        echo 'pkg_postinst_ontarget_${PN}() {' >> $ZOCL_BB
+        echo '  #!/bin/sh' >> $ZOCL_BB
+        echo '  echo "Unloading old XRT Linux kernel modules"' >> $ZOCL_BB
+        echo '  ( rmmod zocl || true ) > /dev/null 2>&1' >> $ZOCL_BB
+        echo '  echo "Loading new XRT Linux kernel modules"' >> $ZOCL_BB
+        echo '  modprobe zocl' >> $ZOCL_BB
+        echo '}' >> $ZOCL_BB
+    fi
+    eval "$SAVED_OPTIONS_LOCAL"
+}
+
+
+
+# --- End internal functions
+
+SAVED_OPTIONS=$(set +o)
+
+# Don't print all commands
+set +x
+# Error on non-zero exit code, by default:
+set -e
+
+# Get real script by read symbol link
+THIS_SCRIPT=`readlink -f ${BASH_SOURCE[0]}`
+
+THIS_SCRIPT_DIR="$( cd "$( dirname "${THIS_SCRIPT}" )" >/dev/null 2>&1 && pwd )"
+
+PROGRAM=`basename $0`
+CONFIG_FILE=""
+PETA_BSP=""
+PROJ_NAME=""
+PLATFROM=""
+XRT_REPO_DIR=`readlink -f ${THIS_SCRIPT_DIR}/..`
+clean=0
+
+while [ $# -gt 0 ]; do
+	case $1 in
+		-help )
+			usage_and_exit 0
+			;;
+		-aarch )
+			shift
+			AARCH=$1
+			;;
+		-clean | clean )
+			clean=1
+			;;
+		--* | -* )
+			error "Unregognized option: $1"
+			;;
+		* )
+			error "Unregognized option: $1"
+			;;
+	esac
+	shift
+done
+
+aarch64_dir="aarch64"
+aarch32_dir="aarch32"
+versal_dir="versal"
+
+if [[ $clean == 1 ]]; then
+    echo $PWD
+    echo "/bin/rm -rf $aarch64_dir $aarch32_dir"
+    /bin/rm -rf $aarch64_dir $aarch32_dir
+    exit 0
+fi
+
+# we pick Petalinux BSP
+if [[ -z ${PETALINUX:+x} ]]; then
+    export PETALINUX="/proj/petalinux/2020.1/petalinux-v2020.1_0217_1/tool/petalinux-v2020.1-final"
```

Comment:
As @chvamshi-xilinx 's comment. Never use hard code path.

Reply:
Please provide a new commit to fix this.

Reply:
as discussed with vamshi, added petalinux.build to control petalinux tool

Reply:
In my mind petalinux.build file is a environment configuration for the script. Instead of check it in to XRT repo, we should ask pipeline guys, who setup the environment, to provide that file. Does this make sense to you?

Reply:
i completely agree with you, but Devops is not ready to set the environment.

Reply:
If this is the case, looks like petalinux.build is a workaround for Devops. Please add this to the workaround list and create a SR to Devops. Once they are ready to do correct thing, we will remove petalinux.build.

Reply:
Please also add that workaround to our Jira epic tracking these. Please contact me offline if you can't find the epic.